components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java

```diff
@@ -113,33 +115,24 @@ public static void process(PublishingFlow publishingFlow) {
     }
 
     /**
-     * Publishes the array list of simplified jsons to Elasticsearch using the Transport client
+     * Publishes the array list of simplified jsons to Elasticsearch using the Java High Level REST Client.
      *
      * @param jsonsToSend array list of json strings to be published to Elasticsearch
-     * @param client      elasticsearch Transport client
+     * @param client      elasticsearch RestHighLevelClient
      */
-    public static void publish(List<String> jsonsToSend, TransportClient client) {
-        try {
-            // Prepares the bulk request
-            BulkRequestBuilder bulkRequest = client.prepareBulk();
-            for (String jsonString : jsonsToSend) {
-                bulkRequest.add(client.prepareIndex("eidata", "data")
-                        .setSource(jsonString, XContentType.JSON)
-                );
-            }
-
-            // Send the bulk request
-            BulkResponse response = bulkRequest.get();
-
-            if (log.isDebugEnabled()) {
-                log.debug("Bulk Request took " + response.getTook() + " milliseconds");
-            }
-        } catch (NoNodeAvailableException e) {
-            log.error("No available Elasticsearch Nodes to connect. Please give correct configurations and" +
-                    " run Elasticsearch.", e);
-        } catch (ElasticsearchSecurityException e) {
-            log.error("Elasticsearch user lacks access to write.", e);
-            client.close();
+    public static void publish(List<String> jsonsToSend, RestHighLevelClient client) {
+
+        // Prepare and Send the bulk request
+        IndexRequest indexRequest = new IndexRequest("eidata");
+        for (String jsonString : jsonsToSend) {
```

Comment:
shouldn't we handle the null case for jsonsToSend array list?

Reply:
Added a null check in 835508f